src/main/scala/se/lu/nateko/cp/data/services/upload/DownloadService.scala

```diff
@@ -128,11 +128,21 @@ class DownloadService(coreConf: MetaCoreConfig, val upload: UploadService, val r
 
 	private def destinyToAuxSourcesFlow(implicit envri: Envri): Flow[FileDestiny, FileEntry, NotUsed] = Flow.apply[FileDestiny]
 		.fold(Vector.empty[FileDestiny])(_ :+ _)
-		.map{dests =>
-			ZipEntry("!TOC.csv") -> destiniesToTocFileSource(dests)
-		}.concat(Source.single(
-			ZipEntry("!LICENCE.pdf") -> licenceSource
-		))
+		.flatMapConcat{dests =>
+			val baseSource = Source.single(ZipEntry("!TOC.csv") -> destiniesToTocFileSource(dests))
+
+			val shouldIncludeLincensePdf = dests.exists(
+				_.obj.references.licence.exists(
+					_.toString().contains("creativecommons.org/licenses/by/4.0")
+				)
+			)
+
+			if (shouldIncludeLincensePdf) {
+				baseSource.concat(Source.single(ZipEntry("!LICENCE.pdf") -> licenceSource))
+			} else {
+				baseSource
+			}
```

Comment:
It would be nice to find a better way to write this, in particular to remove the string comparison. Maybe we could map over the `dests` and do something similar to what is being done in `checkLicenceAcceptance` by only adding the pdf when the license matches one in `mainLicenses`.